pkg/mount/switch_root_linux.go

```diff
@@ -126,32 +126,38 @@ func MoveMount(oldPath string, newPath string) error {
 // 'special' in this context refers to the following non-blockdevice backed
 // mounts that are almost always used: /dev, /proc, /sys, and /run.
 // This function will create the target directories, if necessary.
-// If the target directories already exists, they must be empty.
+// If the target directories already exist, they must be empty.
 // This function skips missing mounts.
 func AddSpecialMounts(newRoot string) error {
 	var mounts = []string{"/dev", "/proc", "/sys", "/run"}
 
 	for _, mount := range mounts {
 		path := filepath.Join(newRoot, mount)
-		// Skip all mounting if the directory does not exists.
+		// Skip all mounting if the directory does not exist.
 		if _, err := os.Stat(mount); os.IsNotExist(err) {
-			fmt.Println("switch_root: Skipping", mount)
+			log.Printf("switch_root: Skipping %q as the dir does not exist", mount)
 			continue
 		} else if err != nil {
 			return err
 		}
+		// Also skip if not currently a mount point
+		if same, err := SameFilesystem("/", mount); err != nil {
+			return err
+		} else if same {
+			log.Printf("switch_root: Skipping %q as it is not a mount", mount)
+			continue
+		}
 		// Make sure the target dir exists and is empty.
-		if _, err := os.Stat(path); os.IsNotExist(err) {
+		fi, err := os.Stat(path)
```

Comment:
replace lines 151-160 with this
if err := os.MkdirAll(path, 0755); err != nil {
return err
}

since MkdirAll does all this stuff anyway.